content/en/docs/ops/integrations/spire/index.md

```diff
@@ -0,0 +1,403 @@
+    {{< /text >}}
+
+1. Register an entry for the SPIRE Agent running on the node:
+
+    {{< text bash >}}
+$ kubectl exec -n spire $SPIRE_SERVER_POD -- \
+/opt/spire/bin/spire-server entry create \
+-spiffeID spiffe://example.org/ns/spire/sa/spire-agent \
+-selector k8s_psat:cluster:demo-cluster \
+-selector k8s_psat:agent_ns:spire \
+-selector k8s_psat:agent_sa:spire-agent \
+-node -socketPath /run/spire/sockets/server.sock
+
+Entry ID         : d38c88d0-7d7a-4957-933c-361a0a3b039c
+SPIFFE ID        : spiffe://example.org/ns/spire/sa/spire-agent
+Parent ID        : spiffe://example.org/spire/server
+Revision         : 0
+TTL              : default
+Selector         : k8s_psat:agent_ns:spire
+Selector         : k8s_psat:agent_sa:spire-agent
+Selector         : k8s_psat:cluster:demo-cluster
+    {{</ text >}}
+
+1. Register an entry for the Ingress-gateway pod:
+
+    {{< text bash >}}
+$ kubectl exec -n spire $`SPIRE_SERVER_POD` -- \
+/opt/spire/bin/spire-server entry create \
+-`spiffeID` `spiffe`://`example.org`/ns/istio-system/sa/`istio-ingressgateway-service-account` \
+-`parentID` `spiffe`://`example.org`/ns/spire/sa/spire-agent \
+-selector k8s:sa:`istio-ingressgateway-service-account` \
+-selector k8s:ns:istio-system \
+-selector k8s:`pod-uid`:$`INGRESS_POD_UID` \
+-`dns` $`INGRESS_POD` \
+-`dns` `istio-ingressgateway`.istio-system.svc \
+-`socketPath` /run/spire/sockets/`server.sock`
+
+Entry ID         : `6f2fe370-5261-4361-ac36-10aae8d91ff7`
+SPIFFE ID        : `spiffe`://`example.org`/ns/istio-system/sa/`istio-ingressgateway-service-account`
+Parent ID        : `spiffe`://`example.org`/ns/spire/sa/spire-agent
+Revision         : 0
+TTL              : default
+Selector         : k8s:ns:istio-system
+Selector         : k8s:`pod-uid`:`63c2bbf5-a8b1-4b1f-ad64-f62ad2a69807`
+Selector         : k8s:sa:`istio-ingressgateway-service-account`
+DNS name         : `istio-ingressgateway`.istio-system.svc
+DNS name         : `istio-ingressgateway-c48554dd6-cff5z`
+    {{</ text >}}
+
+1. Deploy an example workload:
+
+    {{< text bash >}}
+$ istioctl kube-inject --filename @samples/security/spire/sleep-spire.yaml | kubectl apply -f -
+    {{< /text >}}
+
+    Note that the workload will need the SPIFFE CSI Driver volume to access the SPIRE Agent socket. To accomplish this,
+    you can leverage the `spire` pod annotation template from [Install Istio](#install-istio) section or add the CSI volume to
+    the deployment spec of your workload. Both of these alternatives are highlighted on the example snippet below:
+
+    {{< text yaml >}}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: sleep
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: sleep
+  template:
+    metadata:
+      labels:
+        app: sleep
+      # Injects custom sidecar template
+      annotations:
+        inject.istio.io/templates: "sidecar,spire"

```

Comment:
why spire enable or not shall be aware by every pod/deployment through this annotation? Is there a global flag for it ?Should it be enabled in mesh scope?

Reply:
The annotation applies the custom `spire` template created on the configuration of the `Install Istio` section:
```
      templates:
        spire: |
          spec:
            containers:
            - name: istio-proxy
              volumeMounts:
              - name: workload-socket
                mountPath: /run/secrets/workload-spiffe-uds
                readOnly: true
             volumes:
               - name: workload-socket
                 csi:
                   driver: "csi.spiffe.io"
```
The custom template feature facilitates the injection of the `spiffe-csi-driver` volume into workload pods. 

But one can always do it manually by adding the volume to the workload spec:
```
      volumes:
      - name: tmp
        emptyDir: {}
      # CSI volume
      - name: workload-socket
        csi:
          driver: "csi.spiffe.io"
```